packaging/linux/post_install.sh

```diff
@@ -37,6 +33,68 @@ make_mountpoint() {
   fi
 }
 
+# We use machinectl here because systemctl commands do not work in either a su shell or with systemd-run --uid.
+exec_as() {
+    user=$1
+    shift
+    machinectl -q shell "$user"@ "$@"
+}
+
+# Exits with 0 iff the given user is running the service with systemd
+systemd_unit_active_for() {
+    user=$1
+    service=$2
+
+    # exit with 1 if command fails (for example, if not using systemd)
+    active_status="$(exec_as "$user" /usr/bin/systemctl --user is-active "$service")" || return "$?"
+
+    # exit code lost by machinectl, so grep stdout
+    # use -w so "active" needle doesn't match "inactive"
+    echo "$active_status" | grep -wq "active"
+}
+
+systemctl_restart_as() {
+    user=$1
+    service=$2
+    echo "Restarting $service for $user..."
+    exec_as "$user" /usr/bin/systemctl --user restart "$service"
+}
+
+systemd_safe_restart_services() {
+    pids="$(pidof /usr/bin/keybase)"
+    while read -r pid; do
+        user="$(ps -o user= -p "$pid")"
+
+        if systemd_unit_active_for "$user" "kbfs.service"; then
+            # KBFS running, need to check mounts before proceeding
+
+            # Don't use `mount(8)` to find kbfs mounts in case there are non-kbfs fuse systems
+            mount="$(exec_as "$user" /usr/bin/keybase config get -b mountdir | tr -d '\r')"
```

Comment:
Seems like passing `-d` to `config get` here would be a win, even if we already know the daemon is running.  Though maybe there's some race that can happen and it gets a stale value from the config file?

Reply:
I wasn't sure what to do here, I don't think a race would be very common. Since keybase is already running, what's the issue with contacting service? Just for speed?

Reply:
I suppose the service could be in some weird state of alive but not responding though. There are a lot of edge cases here, I'll add error checking to all these commands.

Reply:
Yeah, just speed.  I just don't see a real downside to putting `-d` in.